lib/matplotlib/patches.py

```diff
@@ -369,8 +378,33 @@ def set_color(self, c):
             For setting the edge or face color individually.
         """
         self.set_facecolor(c)
+        self.set_hatchcolor(c)
         self.set_edgecolor(c)
 
+    def _set_hatchcolor(self, color, set_hatchcolor_from_edgecolor=False):
+        if not set_hatchcolor_from_edgecolor:
+            self.set_hatch_color = True
+            if color is None:
+                self.set_hatch_color = False
+                color = mpl.rcParams['hatch.color']
+            alpha = self._alpha if self._fill else 0
+            self._hatch_color = colors.to_rgba(color, alpha)
+            self.stale = True
+
+        if not self.set_hatch_color:
+            self._hatch_color = colors.to_rgba(color, self._alpha)
+
+    def set_hatchcolor(self, color, set_hatchcolor_from_edgecolor=False):
+        """
+        Set the patch hatch color.
+
+        Parameters
+        ----------
+        c : color or None
+        """
+        self._original_hatchcolor = color
+        self._set_hatchcolor(color, set_hatchcolor_from_edgecolor)
```

Comment:
I think the issue with this kinda fallback is that it will never reach the `else` condition because `mpl.rcParams['hatch.color']` is black by default.

Probably, this example will be useful to understand what I'm trying to say:
```py
import matplotlib.pyplot as plt
import numpy as np

width = 0.35
x = np.arange(4) + 1
y_red = [1, 3, 1, 4]
y_blue = [2, 2, 4, 1]

plt.bar(x - 0.5 * width, y_red, width, label="Height of red bars", hatch="////", facecolor=(0, 0, 0, 0), edgecolor="green")

plt.xticks(x)
plt.yticks([0, 1, 2, 3, 4])
plt.legend()
plt.savefig("hatch.png")
plt.show()
```

Output
![image](https://github.com/matplotlib/matplotlib/assets/66066662/e652c645-295a-46d4-93e0-ba7ecf5b5a41)

Expected Output
![image](https://github.com/matplotlib/matplotlib/assets/66066662/514b291a-c8c7-4da0-aec4-d8e0578d0594)

The expected output of the hatchcolor should default to green but will default to black when we use the above logic.
I'm not sure if this is clear enough. Lmk.